src/plugins/discover/public/application/components/sidebar/discover_sidebar.tsx

```diff
@@ -224,6 +229,38 @@ export function DiscoverSidebar({
     return map;
   }, [fields, useNewFieldsApi, selectedFields]);
 
+  const deleteField = useMemo(
+    () =>
+      canEditIndexPatternField
+        ? async (fieldName: string) => {
+            const ref = indexPatternFieldEditor.openDeleteModal({
+              ctx: {
+                // @ts-ignore
```

Comment:
Nit: I'm always scared by `@ts-ignore` as they can shadow future legitimate errors. What about just extending the condition in line 234: `canEditIndexPatternField && selectedIndexPattern`. This will resolve the type error without reducing type safety (like for example if the index pattern type changes in the future or something)